providers/flagd/src/test/java/dev/openfeature/contrib/providers/flagd/FlagdOptionsTest.java

```diff
@@ -64,32 +59,92 @@ public void TestBuilderOptions() {
         assertEquals("app=weatherApp", flagdOptions.getSelector());
         assertEquals("some-path", flagdOptions.getOfflineFlagSourcePath());
         assertEquals(openTelemetry, flagdOptions.getOpenTelemetry());
-        assertEquals(Config.Resolver.IN_PROCESS, flagdOptions.getResolverType());
+        assertEquals(Resolver.IN_PROCESS, flagdOptions.getResolverType());
     }
 
 
     @Test
-    public void testValueProviderForEdgeCase_valid() {
+    void testValueProviderForEdgeCase_valid() {
         Function<String, String> valueProvider = s -> "in-process";
-        assertEquals(Config.Resolver.IN_PROCESS, Config.fromValueProvider(valueProvider));
+        assertEquals(Resolver.IN_PROCESS, Config.fromValueProvider(valueProvider));
 
         valueProvider = s -> "IN-PROCESS";
-        assertEquals(Config.Resolver.IN_PROCESS, Config.fromValueProvider(valueProvider));
+        assertEquals(Resolver.IN_PROCESS, Config.fromValueProvider(valueProvider));
 
         valueProvider = s -> "rpc";
-        assertEquals(Config.Resolver.RPC, Config.fromValueProvider(valueProvider));
+        assertEquals(Resolver.RPC, Config.fromValueProvider(valueProvider));
 
         valueProvider = s -> "RPC";
-        assertEquals(Config.Resolver.RPC, Config.fromValueProvider(valueProvider));
+        assertEquals(Resolver.RPC, Config.fromValueProvider(valueProvider));
     }
 
     @Test
-    public void testValueProviderForEdgeCase_invalid() {
+    void testValueProviderForEdgeCase_invalid() {
         Function<String, String> dummy = s -> "some-other";
         assertEquals(Config.DEFAULT_RESOLVER_TYPE, Config.fromValueProvider(dummy));
 
         dummy = s -> null;
         assertEquals(Config.DEFAULT_RESOLVER_TYPE, Config.fromValueProvider(dummy));
     }
 
+    @Test
+    @Disabled("Currently there is no defined way on how to set environment variables for tests")
```

Comment:
In the past, I used JUnit Pioneer APIs for this: https://javadoc.io/doc/org.junit-pioneer/junit-pioneer/latest/org.junitpioneer/org/junitpioneer/jupiter/SetEnvironmentVariable.html
Looking into other projects (e.g., OTel), they also use this: https://github.com/open-telemetry/opentelemetry-java/blob/934da846f0733fd10872dedfef06019d026ca4d0/api/all/src/test/java/io/opentelemetry/api/internal/ConfigUtilTest.java

Reply:
i love junit pioneer and i can highly recommend it (totally unbiased )  - but i suggest a separate ticket for this

Reply:
Yeah, this is a tricky situation. When I first encountered this issue, I came across `JUnit Pioneer` but given this is a shared repo, I did not add this dep. And I introduced `fromValueProvider` to test. At at the end of the day, we shouldn't worry about internals of  `System::getenv` but only see what our code does when value get retuened to our logic. We could something similar here with private wrapper to `builder()`  (ex:- builder(Function<String, string> supplier)) . But I will leave this decision to you :)

Reply:
created https://github.com/open-feature/java-sdk-contrib/issues/818 to cover this topic, so we can discuss this at as a dedicated topic

Reply:
Let's follow-up on this through #818 so I will resolve the conversation